app/controllers/references_controller.rb

```diff
@@ -1,10 +1,37 @@
 class ReferencesController < ApplicationController
   before_action :ensure_current_user
 
+  # Shame: refactor the hell out of this
   def create
     list = List.find(params[:list_id])
-    paper = Paper.find(params[:reference][:paper_id])
-    Reference.create!(list_id: list.id, paper_id: paper.id)
+
+    # Check if uuid. If not, then ask Pubmed.
+    paper_id = params[:reference][:paper_id]
+    if uuid? paper_id
+      paper = Paper.find(paper_id)
+    else
+      pubmed = Pubmed.new
+      results = pubmed.search paper_id
+      data = results['result'][paper_id]
+      paper = (
+        Paper.find_by(pubmed_id: paper_id) ||
+        Paper.find_by(doi: paper_id) ||
+        # I really hate this due to long experience with it,
+        # but we should probably some sort of mapping
+        Paper.create(
+          pubmed_id: data['uid'],
+          title: data['title'],
+          published_at: data['pubdate'],
+          doi: data['elocationid'].sub(/^doi: /, "")
```

Comment:
The other piece we'll need to add is authors.  i'm not sure if there's rails magic that handles this, but if there's an author name that is already in the db, it should map to the existing uuid.